boskos/janitor/janitor.go

```diff
@@ -0,0 +1,153 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"errors"
+	"fmt"
+	"os/exec"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/Sirupsen/logrus"
+	"k8s.io/test-infra/boskos/client"
+)
+
+var (
+	clean = janitorClean
+)
+
+type semaphore chan bool
+
+// acquire 1 resources
+func (s semaphore) P() error {
+	select {
+	case s <- true:
+	default:
+		return errors.New("channel is full")
+	}
+
+	return nil
+}
+
+// release 1 resources
+func (s semaphore) V() {
+	<-s
+}
+
+func main() {
+	logrus.SetFormatter(&logrus.JSONFormatter{})
+	boskos := client.NewClient("Janitor", "http://boskos")
+	logrus.Infof("[Janitor] Initialized boskos client!")
+	sem := make(semaphore, 10)
```

Comment:
After a second thought, each janitor only takes a few minutes, which does not worth update itself. I'll let reaper clean up stale janitors.